### PR TITLE
Label expression API error response

### PR DESCRIPTION
**Pull request by molnarlaura**

### CODE REVIEW

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/ql/SnomedQueryLabelerRequest.java [hunk 1]
```diff
@@ -73,13 +75,29 @@ public Expressions execute(BranchContext context) {
 		SnomedQuerySerializer querySerializer = context.service(SnomedQuerySerializer.class);
 		final Set<String> conceptIdsToLabel = Sets.newHashSetWithExpectedSize(expressions.size());
 		final Map<String, Query> queries = Maps.newHashMapWithExpectedSize(expressions.size());
+		Map<String, Object> errors = Maps.newHashMap();
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/ql/SnomedQueryLabelerRequest.java]: Use a `LinkedHashMap` to preserve the order of expression in the error response.

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/ql/SnomedQueryLabelerRequest.java [hunk 2]
```diff
@@ -73,13 +75,29 @@ public Expressions execute(BranchContext context) {
...
 		for (String expression : expressions) {
 			if (Strings.isNullOrEmpty(expression)) {
 				continue;
 			}
-			Query query = queries.computeIfAbsent(expression, (key) -> context.service(SnomedQueryParser.class).parse(key));
-			conceptIdsToLabel.addAll(collect(query));
+			try {
+				Query query = queries.computeIfAbsent(expression, (key) -> context.service(SnomedQueryParser.class).parse(key));
+				conceptIdsToLabel.addAll(collect(query));
+			} catch (Exception e) {
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/ql/SnomedQueryLabelerRequest.java]: Catch only ApiExceptions and add the missing else clause which propagates the error to the caller.

File: snomed/com.b2international.snowowl.snomed.datastore.tests/src/com/b2international/snowowl/snomed/core/ql/SnomedQueryLabelerRequestTest.java [hunk 3]
```diff
@@ -202,5 +205,21 @@ public void bulkLabelExpressions() throws Exception {
...
+	public void wrongExpression() throws Exception {
+		String ptUk = "SNOMED CT Concept UK";
+		indexRevision(MAIN,
+				DocumentBuilders.concept(Concepts.ROOT_CONCEPT).preferredDescriptions(ImmutableList.of(
+						new SnomedDescriptionFragment(generateDescriptionId(), Concepts.SYNONYM, ptUk, Concepts.REFSET_LANGUAGE_TYPE_UK)))
+						.build());
+
+		try {
+			bulkLabel(List.of("test1", "test2", Concepts.ROOT_CONCEPT), SnomedConcept.Expand.PREFERRED_TERM, ImmutableList.of(ExtendedLocale.valueOf("en-x-" + Concepts.REFSET_LANGUAGE_TYPE_US)));
+		} catch (Exception e) {
```
Comment (cmark) [inline snomed/com.b2international.snowowl.snomed.datastore.tests/src/com/b2international/snowowl/snomed/core/ql/SnomedQueryLabelerRequestTest.java]: Either use JUnit 4's appropriate Rule implementation to catch and assert thrown exception type and message or use assertj's [exception assertion](https://assertj.github.io/doc/#assertj-core-exception-assertions).

**Review by cmark — CHANGES_REQUESTED** [review]

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/ql/SnomedQueryLabelerRequest.java (hunk 2 above)
Comment (molnarlaura) [reply snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/ql/SnomedQueryLabelerRequest.java]: Done

File: snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/ql/SnomedQueryLabelerRequest.java (hunk 1 above)
Comment (molnarlaura) [reply snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/core/ql/SnomedQueryLabelerRequest.java]: Done

File: snomed/com.b2international.snowowl.snomed.datastore.tests/src/com/b2international/snowowl/snomed/core/ql/SnomedQueryLabelerRequestTest.java (hunk 3 above)
Comment (molnarlaura) [reply snomed/com.b2international.snowowl.snomed.datastore.tests/src/com/b2international/snowowl/snomed/core/ql/SnomedQueryLabelerRequestTest.java]: Done

**Review by cmark — APPROVED** [review]
LGTM!

**Pull request merged**